_includes/app/util.clj

```diff
@@ -0,0 +1,87 @@
+(ns test.util
+  (:require   [clojure.java.jdbc :as j]
+              [clojure.walk :as walk]))
+
+(defn txn-restart-err-p
+  "Takes an exception and return true if it is a CockroachDB retry error."
+  [e]
+  (when-let [m (.getMessage e)]
+    (condp instance? e
+      java.sql.BatchUpdateException
+      (and (re-find #"getNextExc" m)
+           (txn-restart-err-p (.getNextException e)))
+
+      org.postgresql.util.PSQLException
+      (or (re-find  #"restart transaction" m) (re-find #"force_retry" m))
```

Comment:
Can we use `getSQLState` (which I think should return one of the 5-character error codes) instead of matching on `getMessage`?